packages/utilities/psammead-calendars/src/calendars/jalaali.test.js

```diff
@@ -0,0 +1,115 @@
+import moment from 'moment';
+import jalaali from './jalaali';
+
+describe('Jalaali Conversion Tests', () => {
+  moment.defineLocale('ps', {});
+  const dateTestScenarios = [
+    {
+      date: '2019-01-01',
+      locale: 'fa',
+      expected: '11 دی 1397',
+      summary:
+        'should return first day of the year 2019 in Jalaali for persian',
+    },
+    {
+      date: '2019-12-31',
+      locale: 'fa',
+      expected: '10 دی 1398',
+      summary: 'should return last day of the year 2019 in Jalaali for persian',
+    },
+    {
+      date: '2025-02-01',
+      locale: 'fa',
+      expected: '13 بهمن 1403',
+      summary: 'should return first day of Febuary 2025 in Jalaali for persian',
+    },
+    {
+      date: '2025-11-31',
+      locale: 'fa',
+      expected: null,
+      summary: "should return null as date doesn't exist",
+    },
+    {
+      date: '31-2019-12',
+      locale: 'fa',
+      expected: null,
+      summary: 'should return null as this is an invalid moment',
+    },
+    {
+      date: '2019-01-01',
+      locale: 'ps',
+      expected: '11 مرغومی 1397',
+      summary: 'should return first day of the year 2019 in Jalaali for pashto',
+    },
+    {
+      date: '2019-12-31',
+      locale: 'ps',
+      expected: '10 مرغومی 1398',
+      summary: 'should return last day of the year 2019 in Jalaali for pashto',
+    },
+    {
+      date: '2025-02-01',
+      locale: 'ps',
+      expected: '13 سلواغه 1403',
+      summary: 'should return first day of Febuary 2025 in Jalaali for pashto',
+    },
+    {
+      date: '2025-11-31',
+      locale: 'ps',
+      expected: null,
+      summary: "should return null as date doesn't exist",
+    },
+    {
+      date: '31-2019-12',
+      locale: 'ps',
+      expected: null,
+      summary: 'should return null as this is an invalid moment',
+    },
+  ];
+  const nonMomentTestScenarios = [
```

Comment:
It's great to see you've chosen to separate these scenarios.

As they don't relate to Moment, would it be better to use a name that doesn't have "moment" in it and rename `dateTestScenarios` to `momentTestScenarios`? It might make things more readable.

Reply:
What do you suggest calling it?

Reply:
Would `momentTestScenarios` and `formatDateTestScenarios` make sense?

Reply:
They both use `formatDate()` so I'm not sure that's the best wording

Reply:
That's a good argument _against_ separating the scenarios. Perhaps this would be a better route to keeping the test pure? https://gist.github.com/simonsinclair/6f37753e74050339edfa63cf0ef408db